symbolic-symcache/src/preamble.rs

```diff
@@ -0,0 +1,73 @@
+use std::mem;
+
+use crate::{SymCacheError, SymCacheErrorKind};
+
+/// The magic file preamble as individual bytes.
+pub const SYMCACHE_MAGIC: [u8; 4] = *b"SYMC";
+
+/// The start of a SymCache file.
+#[repr(C, packed)]
+#[derive(Default, Copy, Clone, Debug, PartialEq, Eq)]
+pub struct Preamble {
+    /// Magic bytes, see `SYMCACHE_MAGIC`.
+    pub magic: [u8; 4],
+    /// Version of the SymCache file format.
+    pub version: u32,
+}
+
+impl Preamble {
+    pub(crate) fn parse(buf: &[u8]) -> Result<Self, SymCacheError> {
+        let preamble_size = mem::size_of::<Self>();
+
+        if buf.len() < preamble_size {
+            return Err(SymCacheErrorKind::BadFileHeader.into());
+        }
+        // SAFETY: we checked that the buffer is well aligned and large enough to fit a `Preamble`.
```

Comment:
maybe not "well aligned". But `#[repr(packed)]` essentially removes all alignment constraints.